packages/grid/src/vaadin-grid.d.ts

```diff
@@ -299,29 +299,41 @@ export interface GridEventMap<TItem> extends HTMLElementEventMap, GridCustomEven
  *
  * The following shadow DOM parts are available for styling:
  *
- * Part name              | Description
- * -----------------------|----------------
- * `row`                  | Row in the internal table
- * `expanded-row`         | Expanded row
- * `selected-row`         | Selected row
- * `details-opened-row`   | Row with details open
- * `odd-row`              | Odd row
- * `first-row`            | The first body row
- * `last-row`             | The last body row
- * `dragstart-row`        | Set on the row for one frame when drag is starting. The value is a number when multiple rows are dragged
- * `dragover-above-row`   | Set on the row when the a row is dragged over above
- * `dragover-below-row`   | Set on the row when the a row is dragged over below
- * `dragover-on-top-row`  | Set on the row when the a row is dragged over on top
- * `drag-disabled-row`    | Set to a row that isn't available for dragging
- * `drop-disabled-row`    | Set to a row that can't be dropped on top of
- * `cell`                 | Cell in the internal table
- * `header-cell`          | Header cell in the internal table
- * `body-cell`            | Body cell in the internal table
- * `footer-cell`          | Footer cell in the internal table
- * `details-cell`         | Row details cell in the internal table
- * `focused-cell`         | Focused cell in the internal table
- * `resize-handle`        | Handle for resizing the columns
- * `reorder-ghost`        | Ghost element of the header cell being dragged
+ * Part name                  | Description
+ * ---------------------------|----------------
+ * `row`                      | Row in the internal table
+ * `expanded-row`             | Expanded row
+ * `selected-row`             | Selected row
+ * `details-opened-row`       | Row with details open
+ * `odd-row`                  | Odd row
+ * `first-row`                | The first body row
+ * `last-row`                 | The last body row
+ * `dragstart-row`            | Set on the row for one frame when drag is starting. The value is a number when multiple rows are dragged
+ * `dragover-above-row`       | Set on the row when the a row is dragged over above
+ * `dragover-below-row`       | Set on the row when the a row is dragged over below
+ * `dragover-on-top-row`      | Set on the row when the a row is dragged over on top
+ * `drag-disabled-row`        | Set to a row that isn't available for dragging
+ * `drop-disabled-row`        | Set to a row that can't be dropped on top of
```

Comment:
Considering that many of the corresponding row state attributes are used to style cells in the theme, we probably want to expose all row state attributes as part names on the cell.

For example odd rows:
https://github.com/vaadin/web-components/blob/54f903fedd66ff5a9d04054d6d9714d0cceff4e0/packages/grid/theme/lumo/vaadin-grid-styles.js#L310-L316

Or drag over styles:
https://github.com/vaadin/web-components/blob/54f903fedd66ff5a9d04054d6d9714d0cceff4e0/packages/grid/theme/lumo/vaadin-grid-styles.js#L114-L124

Also interesting is that we often use `:not` in the theme, which is not possible with `::part` (I think?).

Reply:
Good point, thanks. I will look into adding more parts e.g. `odd-row-cell` etc.

Reply:
Added more cell parts for these states, please check the recent commits in the branch.